File: 20A40/cir-villalobos/e32/Gato.cpp
#include "Gato.h"
#include <string>

Gato::Gato(){}

Gato::Gato(
    int nPatas, std::string& colorP,std::string& nombreP, 
    std::string& especieP, int ratAtrap, int araña) : 
        Mamifero(nPatas, colorP, nombreP, especieP),
        ratonesAtrapados(ratAtrap), arañados(araña){}
Gato::~Gato(){}

bool Gato::setRatones(int ratones){
    if(!(ratones >= 0)) return false;
    this->ratonesAtrapados = ratones;
}

bool Gato::setArañados(int arañadosP){
    if(!(arañadosP >= 0)) return false;
    this->arañados = arañadosP;
}

int Gato::getRatones() {    return ratonesAtrapados;    }
int Gato::getArañados(){    return arañados;            }

File: 20A40/cir-Israel/Ejercicio 32/Gato.java
public class Gato extends Mamifero {
  private byte ratonesAtrapados;
  private byte arañados;

  public Gato() {}

  public Gato(String especie) {
      super(especie, 4, "gris");
      ratonesAtrapados = 0;
      arañados = 0;
  }

  public Gato(String especie, int patas, String color, byte ratones, byte arañados) {
    super(especie, patas, color);
    setRatonesAtrapados(ratones);
    setArañados(arañados);
  }

  public boolean setRatonesAtrapados(byte ratones) {
      if (ratones > 0) {
          ratonesAtrapados = ratones;
          return true;
      } else return false;
  }

  public boolean setArañados(byte arañados) {
      if (arañados > 0) {
          this.arañados = arañados;
          return true;
      } else return false;
  }

  public byte getRatonesAtrapados() { return ratonesAtrapados; }
  public byte getArañados() { return arañados; }
}